HTTP/1.1 200 OK
Transfer-Encoding: chunked
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Cache-Control: no-cache
Content-Security-Policy: default-src 'none'
Content-Type: application/json; charset=utf-8
Date: Sat, 24 Mar 2018 12:30:59 GMT
Link: <https://api.github.com/search/issues?q=repo%3Aemberjs%2Fember.js+type%3Aissue+state%3Aclosed&page=2>; rel="next", <https://api.github.com/search/issues?q=repo%3Aemberjs%2Fember.js+type%3Aissue+state%3Aclosed&page=34>; rel="last"
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Server: GitHub.com
Status: 200 OK
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Accepted-Oauth-Scopes: 
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-Github-Media-Type: github.v3; format=json
X-Github-Request-Id: D07B:5F56:32D1B5D:6D5F411:5AB64503
X-Oauth-Scopes: 
X-Ratelimit-Limit: 30
X-Ratelimit-Remaining: 24
X-Ratelimit-Reset: 1521894679
X-Runtime-Rack: 0.379987
X-Xss-Protection: 1; mode=block

8000
{"total_count":5291,"incomplete_results":false,"items":[{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16371","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16371/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16371/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16371/events","html_url":"https://github.com/emberjs/ember.js/issues/16371","id":304753186,"number":16371,"title":"Tutorial: Integration test fails on triggerKeyEvent().","user":{"login":"rayascott","id":30600616,"avatar_url":"https://avatars0.githubusercontent.com/u/30600616?v=4","gravatar_id":"","url":"https://api.github.com/users/rayascott","html_url":"https://github.com/rayascott","followers_url":"https://api.github.com/users/rayascott/followers","following_url":"https://api.github.com/users/rayascott/following{/other_user}","gists_url":"https://api.github.com/users/rayascott/gists{/gist_id}","starred_url":"https://api.github.com/users/rayascott/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rayascott/subscriptions","organizations_url":"https://api.github.com/users/rayascott/orgs","repos_url":"https://api.github.com/users/rayascott/repos","events_url":"https://api.github.com/users/rayascott/events{/privacy}","received_events_url":"https://api.github.com/users/rayascott/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-13T12:55:58Z","updated_at":"2018-03-13T18:48:41Z","closed_at":"2018-03-13T18:48:41Z","author_association":"NONE","body":"From [Autocomplete Component](https://guides.emberjs.com/v3.0.0/tutorial/autocomplete-component/) under the heading \"An Integration Test\" (and just before the heading \"Application Tests\") the test in file `tests/integration/components/list-filter-test.js` (\"should update with matching listings\") fails due to no element passed to `triggerKeyEvent()` call. This failure also applies to the test \"should initially load all listings\".\r\n\r\nAlthough I've only looked at Ember for less than 2 days, I get the impression this error is caused by the fact that class `.list-filter` does not exist on the `input` HTML tag generated by the `list-filter` component.  ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16370","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16370/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16370/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16370/events","html_url":"https://github.com/emberjs/ember.js/issues/16370","id":304746969,"number":16370,"title":"Tutorial: this.on referent does not exist.","user":{"login":"rayascott","id":30600616,"avatar_url":"https://avatars0.githubusercontent.com/u/30600616?v=4","gravatar_id":"","url":"https://api.github.com/users/rayascott","html_url":"https://github.com/rayascott","followers_url":"https://api.github.com/users/rayascott/followers","following_url":"https://api.github.com/users/rayascott/following{/other_user}","gists_url":"https://api.github.com/users/rayascott/gists{/gist_id}","starred_url":"https://api.github.com/users/rayascott/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rayascott/subscriptions","organizations_url":"https://api.github.com/users/rayascott/orgs","repos_url":"https://api.github.com/users/rayascott/repos","events_url":"https://api.github.com/users/rayascott/events{/privacy}","received_events_url":"https://api.github.com/users/rayascott/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-13T12:36:34Z","updated_at":"2018-03-13T18:44:40Z","closed_at":"2018-03-13T18:44:40Z","author_association":"NONE","body":"From the [Autocomplete Component step](https://guides.emberjs.com/v3.0.0/tutorial/autocomplete-component/) under the heading \"An Integration Test\" after the first code example, the reference `this.on` has no referent. \r\n\r\nUnless I'm misreading the text, or it's down to my lack of knowledge regarding JavaScript & Ember development, I can't find where the `this.on` call is used. Is this a documentation bug? ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16369","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16369/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16369/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16369/events","html_url":"https://github.com/emberjs/ember.js/issues/16369","id":304743692,"number":16369,"title":"Tutorial: autocomplete component example documentation bug","user":{"login":"rayascott","id":30600616,"avatar_url":"https://avatars0.githubusercontent.com/u/30600616?v=4","gravatar_id":"","url":"https://api.github.com/users/rayascott","html_url":"https://github.com/rayascott","followers_url":"https://api.github.com/users/rayascott/followers","following_url":"https://api.github.com/users/rayascott/following{/other_user}","gists_url":"https://api.github.com/users/rayascott/gists{/gist_id}","starred_url":"https://api.github.com/users/rayascott/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rayascott/subscriptions","organizations_url":"https://api.github.com/users/rayascott/orgs","repos_url":"https://api.github.com/users/rayascott/repos","events_url":"https://api.github.com/users/rayascott/events{/privacy}","received_events_url":"https://api.github.com/users/rayascott/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-03-13T12:25:56Z","updated_at":"2018-03-14T17:02:01Z","closed_at":"2018-03-13T18:41:27Z","author_association":"NONE","body":"From the [Autocomplete Component step](https://guides.emberjs.com/v3.0.0/tutorial/autocomplete-component/) under the heading \"An Integration Test\" the **second** code listing for `tests/integration/components/list-filter-test.js` (**19** lines of code), line 13 should be highlighted as green, because the `async` keyword has been added to the `function` declaration. Without it, the test fails once the `await` keyword is added in the next code example.  ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16368","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16368/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16368/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16368/events","html_url":"https://github.com/emberjs/ember.js/issues/16368","id":304682572,"number":16368,"title":"Tutorial: Tests Fail With 404 on /api/rentals after Ember Data step","user":{"login":"rayascott","id":30600616,"avatar_url":"https://avatars0.githubusercontent.com/u/30600616?v=4","gravatar_id":"","url":"https://api.github.com/users/rayascott","html_url":"https://github.com/rayascott","followers_url":"https://api.github.com/users/rayascott/followers","following_url":"https://api.github.com/users/rayascott/following{/other_user}","gists_url":"https://api.github.com/users/rayascott/gists{/gist_id}","starred_url":"https://api.github.com/users/rayascott/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rayascott/subscriptions","organizations_url":"https://api.github.com/users/rayascott/orgs","repos_url":"https://api.github.com/users/rayascott/repos","events_url":"https://api.github.com/users/rayascott/events{/privacy}","received_events_url":"https://api.github.com/users/rayascott/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-13T09:17:42Z","updated_at":"2018-03-13T18:47:32Z","closed_at":"2018-03-13T18:47:32Z","author_association":"NONE","body":"I've been following the official Ember tutorial and have been writing the recommended tests along with the app code. After completing the [Ember Data step](https://guides.emberjs.com/v3.0.0/tutorial/ember-data/) I noticed that all of my acceptance tests were failing with a 404 on the GET request to /api/rentals. I am not seeing this issue with the app code however. The rentals display correctly on the rentals page.\r\n\r\nCan anyone with some Ember experience shed some light on what's wrong and/or how to fix it? Thanks.\r\n\r\n\r\n\r\n> Promise rejected during \"should show rentals as the home page\": Ember Data Request GET /api/rentals returned a 404\r\nPayload (Empty Content-Type)\r\nNot found: /api/rentals@ 381 ms\r\nSource: \t\r\nError: Ember Data Request GET /api/rentals returned a 404\r\nPayload (Empty Content-Type)\r\nNot found: /api/rentals\r\n    at ErrorClass.EmberError (http://localhost:7357/assets/vendor.js:24125:25)\r\n    at ErrorClass.AdapterError (http://localhost:7357/assets/vendor.js:157167:15)\r\n    at new ErrorClass (http://localhost:7357/assets/vendor.js:157185:22)\r\n    at Class.handleResponse (http://localhost:7357/assets/vendor.js:169227:18)\r\n    at ajaxError (http://localhost:7357/assets/vendor.js:169720:25)\r\n    at Class.hash.error (http://localhost:7357/assets/vendor.js:169308:23)\r\n    at fire (http://localhost:7357/assets/vendor.js:3607:31)\r\n    at Object.fireWith [as rejectWith] (http://localhost:7357/assets/vendor.js:3737:7)\r\n    at done (http://localhost:7357/assets/vendor.js:9646:14)\r\n    at XMLHttpRequest.<anonymous> (http://localhost:7357/assets/vendor.js:9887:9)","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16364","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16364/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16364/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16364/events","html_url":"https://github.com/emberjs/ember.js/issues/16364","id":304303019,"number":16364,"title":"`Ember.set` does not return the value returned by the setter of a computed value","user":{"login":"dennis95stumm","id":6422711,"avatar_url":"https://avatars2.githubusercontent.com/u/6422711?v=4","gravatar_id":"","url":"https://api.github.com/users/dennis95stumm","html_url":"https://github.com/dennis95stumm","followers_url":"https://api.github.com/users/dennis95stumm/followers","following_url":"https://api.github.com/users/dennis95stumm/following{/other_user}","gists_url":"https://api.github.com/users/dennis95stumm/gists{/gist_id}","starred_url":"https://api.github.com/users/dennis95stumm/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/dennis95stumm/subscriptions","organizations_url":"https://api.github.com/users/dennis95stumm/orgs","repos_url":"https://api.github.com/users/dennis95stumm/repos","events_url":"https://api.github.com/users/dennis95stumm/events{/privacy}","received_events_url":"https://api.github.com/users/dennis95stumm/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-12T09:52:24Z","updated_at":"2018-03-13T06:41:18Z","closed_at":"2018-03-13T06:41:18Z","author_association":"NONE","body":"When setting a value with `Ember.set` on computed values the method does not return the value\r\nreturned by the setter of the computed property. Instead of that the passed value gets returned. See [this twiddle](https://ember-twiddle.com/5c245207e14b55c57f70527d8324b0a4?openFiles=controllers.application.js%2C) for more details.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16361","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16361/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16361/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16361/events","html_url":"https://github.com/emberjs/ember.js/issues/16361","id":304106626,"number":16361,"title":"Conflict between JS module API and Module Unification","user":{"login":"tomdale","id":90888,"avatar_url":"https://avatars2.githubusercontent.com/u/90888?v=4","gravatar_id":"","url":"https://api.github.com/users/tomdale","html_url":"https://github.com/tomdale","followers_url":"https://api.github.com/users/tomdale/followers","following_url":"https://api.github.com/users/tomdale/following{/other_user}","gists_url":"https://api.github.com/users/tomdale/gists{/gist_id}","starred_url":"https://api.github.com/users/tomdale/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/tomdale/subscriptions","organizations_url":"https://api.github.com/users/tomdale/orgs","repos_url":"https://api.github.com/users/tomdale/repos","events_url":"https://api.github.com/users/tomdale/events{/privacy}","received_events_url":"https://api.github.com/users/tomdale/received_events","type":"User","site_admin":false},"labels":[{"id":863470709,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Module%20Unification","name":"Module Unification","color":"db087c","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-03-10T21:25:06Z","updated_at":"2018-03-23T15:43:15Z","closed_at":"2018-03-23T15:43:15Z","author_association":"OWNER","body":"I discovered an issue while pairing with @wycats that I wanted to write down for future discussion.\r\n\r\nWe were upgrading an Ember app to use Module Unification that had already adopted the JS module API. The migrator produced invalid code for helpers due to the following conflict:\r\n\r\n1. Helper functions must be wrapped with the `helper` function exported by `@ember/component/helper`.\r\n2. In Module Unification, helpers must be a named export called `helper`.\r\n\r\nThe migrator produces code that looks like this:\r\n\r\n```js\r\nimport { helper } from '@ember/components/helper';\r\nfunction myHelper() { /* ... */ }\r\n// Error: cannot redeclare `helper`\r\nexport const helper = helper(myHelper);\r\n```\r\n\r\nOne way to fix this is for the migrator to detect this conflict and rename the import (since the export name cannot be changed), e.g. `import { helper as emberHelper } from '@ember/components/helper'`. But this is something that everyone will have to type, so we may want to think about this conflict and if there are design changes we want to consider.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16345","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16345/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16345/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16345/events","html_url":"https://github.com/emberjs/ember.js/issues/16345","id":303678453,"number":16345,"title":"Inject header for static content requests","user":{"login":"camarrone","id":11856252,"avatar_url":"https://avatars1.githubusercontent.com/u/11856252?v=4","gravatar_id":"","url":"https://api.github.com/users/camarrone","html_url":"https://github.com/camarrone","followers_url":"https://api.github.com/users/camarrone/followers","following_url":"https://api.github.com/users/camarrone/following{/other_user}","gists_url":"https://api.github.com/users/camarrone/gists{/gist_id}","starred_url":"https://api.github.com/users/camarrone/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/camarrone/subscriptions","organizations_url":"https://api.github.com/users/camarrone/orgs","repos_url":"https://api.github.com/users/camarrone/repos","events_url":"https://api.github.com/users/camarrone/events{/privacy}","received_events_url":"https://api.github.com/users/camarrone/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-09T00:08:07Z","updated_at":"2018-03-09T00:50:07Z","closed_at":"2018-03-09T00:50:07Z","author_association":"NONE","body":"Hello everyone.\r\n\r\nNot actually an issue but some help needed.\r\n\r\nIs there any way to get ember to inject my custom headers in all requests, even for the `img` tags?\r\n\r\nMy application gets the images path from the database and then uses those path as the `img src` in the templates. For the routes that are used by ember-data, I can send and get the custom headers in the backend dynamic routes but I can't get it in the static content when accessing the image files.\r\n\r\nI'm using expressjs as my backend server.\r\n\r\nHow can this be accomplished?\r\n\r\nThanks very much for the help!","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16344","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16344/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16344/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16344/events","html_url":"https://github.com/emberjs/ember.js/issues/16344","id":303670951,"number":16344,"title":"ember-source@3 + ember-cli@3 does not generate valid tests","user":{"login":"nathanhammond","id":20542,"avatar_url":"https://avatars1.githubusercontent.com/u/20542?v=4","gravatar_id":"","url":"https://api.github.com/users/nathanhammond","html_url":"https://github.com/nathanhammond","followers_url":"https://api.github.com/users/nathanhammond/followers","following_url":"https://api.github.com/users/nathanhammond/following{/other_user}","gists_url":"https://api.github.com/users/nathanhammond/gists{/gist_id}","starred_url":"https://api.github.com/users/nathanhammond/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/nathanhammond/subscriptions","organizations_url":"https://api.github.com/users/nathanhammond/orgs","repos_url":"https://api.github.com/users/nathanhammond/repos","events_url":"https://api.github.com/users/nathanhammond/events{/privacy}","received_events_url":"https://api.github.com/users/nathanhammond/received_events","type":"User","site_admin":false},"labels":[{"id":187302,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Bug","name":"Bug","color":"e10c02","default":false},{"id":394458286,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Has%20PR","name":"Has PR","color":"c5def5","default":false},{"id":259289356,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Help%20Wanted","name":"Help Wanted","color":"009800","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-03-08T23:30:52Z","updated_at":"2018-03-09T20:20:34Z","closed_at":"2018-03-09T20:20:34Z","author_association":"MEMBER","body":"As of https://github.com/ember-cli/ember-cli/pull/7546 landing, [we no longer have any helpers](https://github.com/ember-cli/ember-cli/tree/v3.0.0/blueprints/app/files/tests/helpers) in Ember CLI.\r\n\r\nThis breaks the blueprints for [instance-initializers](https://github.com/emberjs/ember.js/blob/master/blueprints/instance-initializer-test/qunit-rfc-232-files/tests/unit/instance-initializers/__name__-test.js) and [initializers](https://github.com/emberjs/ember.js/blob/master/blueprints/initializer-test/qunit-rfc-232-files/tests/unit/initializers/__name__-test.js).\r\n\r\nIt also is broken for Mocha acceptance tests:\r\nhttps://github.com/emberjs/ember.js/blob/master/blueprints/acceptance-test/mocha-files/tests/acceptance/__name__-test.js\r\n\r\n(I'm not sure that I've enumerated everywhere that it fails.)\r\n\r\nThis bug is effectively a duplicate of https://github.com/emberjs/ember.js/issues/16180, but exists because https://github.com/emberjs/ember.js/pull/16229 is incomplete.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16337","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16337/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16337/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16337/events","html_url":"https://github.com/emberjs/ember.js/issues/16337","id":303385245,"number":16337,"title":"HTMLbars: Cannot pass parameter array to loc helper","user":{"login":"schlesingermatthias","id":20062071,"avatar_url":"https://avatars0.githubusercontent.com/u/20062071?v=4","gravatar_id":"","url":"https://api.github.com/users/schlesingermatthias","html_url":"https://github.com/schlesingermatthias","followers_url":"https://api.github.com/users/schlesingermatthias/followers","following_url":"https://api.github.com/users/schlesingermatthias/following{/other_user}","gists_url":"https://api.github.com/users/schlesingermatthias/gists{/gist_id}","starred_url":"https://api.github.com/users/schlesingermatthias/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/schlesingermatthias/subscriptions","organizations_url":"https://api.github.com/users/schlesingermatthias/orgs","repos_url":"https://api.github.com/users/schlesingermatthias/repos","events_url":"https://api.github.com/users/schlesingermatthias/events{/privacy}","received_events_url":"https://api.github.com/users/schlesingermatthias/received_events","type":"User","site_admin":false},"labels":[{"id":232103760,"url":"https://api.github.com/repos/emberjs/ember.js/labels/wontfix","name":"wontfix","color":"000000","default":true}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-03-08T07:44:15Z","updated_at":"2018-03-08T11:53:05Z","closed_at":"2018-03-08T11:52:47Z","author_association":"NONE","body":"According to the [documentation](http://www.emberjs.com.cn/api/classes/Ember.String.html#method_loc) it should be possible to pass an array of parameters to the loc helper. While this works in JavaScript I was trying to do something similar in a template file. \r\n\r\nSo for a parameterized localization string like this one: \r\n`Hello %@1, today is %@2 `\r\n\r\nI'd like to use it like: \r\n`{{loc \"greetings\" myArray}}`\r\n\r\nUnfortunately this results in\r\n`Hello , today is `\r\n\r\n`{{loc \"greetings\" \"John\" \"Monday\"}}`\r\nworks, but not with a dynamically created array. \r\n\r\nWould it be possible to implement a behaviour similar to the component helper by using a params property, like: \r\n`{{loc \"greetings\" params=myArray}}`","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16334","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16334/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16334/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16334/events","html_url":"https://github.com/emberjs/ember.js/issues/16334","id":303115090,"number":16334,"title":".trigger() from Evented Mixin always fires queued listeners in reverse order","user":{"login":"gfmartinez","id":14170693,"avatar_url":"https://avatars0.githubusercontent.com/u/14170693?v=4","gravatar_id":"","url":"https://api.github.com/users/gfmartinez","html_url":"https://github.com/gfmartinez","followers_url":"https://api.github.com/users/gfmartinez/followers","following_url":"https://api.github.com/users/gfmartinez/following{/other_user}","gists_url":"https://api.github.com/users/gfmartinez/gists{/gist_id}","starred_url":"https://api.github.com/users/gfmartinez/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/gfmartinez/subscriptions","organizations_url":"https://api.github.com/users/gfmartinez/orgs","repos_url":"https://api.github.com/users/gfmartinez/repos","events_url":"https://api.github.com/users/gfmartinez/events{/privacy}","received_events_url":"https://api.github.com/users/gfmartinez/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-03-07T14:16:37Z","updated_at":"2018-03-09T18:19:48Z","closed_at":"2018-03-09T18:05:44Z","author_association":"NONE","body":"https://github.com/emberjs/ember.js/blob/ce3324570a9d6f1ee6525d53cdbf42c1dcaad345/packages/ember-metal/lib/events.js#L130\r\n\r\nThe flattened actions always loop in reverse order from the queued order of listeners.\r\n\r\nI see in the note that this was intended for events registered as ONCE. But use of `.on()` or `.one()` to register an event results in the same reverse order call for queued events.\r\n\r\nSuggest a check before the calling of queued listeners as to direction: reverse or forward. Unless specified, I would expect the listeners to be called in the order in which they were registered.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16328","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16328/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16328/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16328/events","html_url":"https://github.com/emberjs/ember.js/issues/16328","id":302730138,"number":16328,"title":"Tests which pass locally are failing on Heroku","user":{"login":"Charlie-Williams","id":32263035,"avatar_url":"https://avatars3.githubusercontent.com/u/32263035?v=4","gravatar_id":"","url":"https://api.github.com/users/Charlie-Williams","html_url":"https://github.com/Charlie-Williams","followers_url":"https://api.github.com/users/Charlie-Williams/followers","following_url":"https://api.github.com/users/Charlie-Williams/following{/other_user}","gists_url":"https://api.github.com/users/Charlie-Williams/gists{/gist_id}","starred_url":"https://api.github.com/users/Charlie-Williams/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Charlie-Williams/subscriptions","organizations_url":"https://api.github.com/users/Charlie-Williams/orgs","repos_url":"https://api.github.com/users/Charlie-Williams/repos","events_url":"https://api.github.com/users/Charlie-Williams/events{/privacy}","received_events_url":"https://api.github.com/users/Charlie-Williams/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-03-06T14:49:02Z","updated_at":"2018-03-07T16:35:55Z","closed_at":"2018-03-07T16:34:33Z","author_association":"NONE","body":"I am having an issue with Ember/Heroku and I cannot identify exactly where the problems are coming from.\r\n\r\nSpecifically, the qunit tests are breaking for many routes which use a mixin to send data to Adobe Analytics. The behaviour is exactly as expected but when I was running the tests I got `ReferenceError: dataTracker is not defined`, where `dataTracker` comes from a JS library and significant numbers of tests failing.\r\n\r\nLocally, I can solve the issue by wrapping the section of code which calls `dataTracker` in:\r\n```\r\nif (!testing) {\r\n...\r\n}\r\n```\r\nSo now the Mixin looks like this:\r\n```\r\n/* global dataTracker */\r\nimport Mixin from '@ember/object/mixin';\r\nimport Ember from 'ember';\r\n\r\nconst { testing } = Ember;\r\n\r\nexport default Mixin.create({\r\n notifyAdobeAnalytics() {\r\n  if(!testing) {\r\n   let payload = someJSON\r\n\t\r\n   dataTracker.trackEvent('newPage', payload);\r\n   }\r\n  }\r\n});\r\n```\r\n\r\nThis fixes the (many!) broken tests and the behaviour is unaffected... Perfect!\r\n\r\nHowever, when running the same tests in the Heroku staging environment, the same error message is thrown as I was previously getting locally, before adding the `if(!testing)`.\r\n\r\nElsewhere I have looked in our codebase, using the `testing` constant is working normally with Heroku elsewhere so I am unsure as to why it is breaking here. I cannot find any reason why on Heroku I am getting the same error as has been fixed locally.\r\n\r\nAll error logs are of the structure:\r\n```\r\nReferenceError: dataTracker is not defined\r\n    at Class.notifyAdobeAnalytics (http://localhost:4200/assets/hivebench-frontend/mixins/adobe-analytics.js:50:1)\r\n    at Class.didTransition (http://localhost:4200/assets/hivebench-frontend/mixins/adobe-analytics.js:24:1)\r\n    at Router$1.triggerEvent (http://localhost:4200/assets/ember-routing/system/router.js:1023:1)\r\n    at trigger (http://localhost:4200/assets/router.js:111:1)\r\n    at finalizeTransition (http://localhost:4200/assets/router.js:2018:1)\r\n    at http://localhost:4200/assets/router.js:1402:1\r\n    at tryCatcher (http://localhost:4200/assets/rsvp.js:215:1)\r\n    at invokeCallback (http://localhost:4200/assets/rsvp.js:393:1)\r\n    at publish (http://localhost:4200/assets/rsvp.js:379:1)\r\n    at http://localhost:4200/assets/ember-testing/ext/rsvp.js:14:1\r\n```\r\n\r\nI have tried adding it as a global variable by putting `/* global dataTracker */` at the top of the page and various other things but nothing has worked so far.\r\n\r\nThe only issues I can find which look  _vaguely_ related are these:\r\nhttps://github.com/emberjs/ember-qunit/issues/297#issuecomment-352385666\r\nhttps://github.com/emberjs/ember.js/issues/16006\r\nBut they do not have the same problem as I am having.\r\n\r\nI am using Ember version 2.16.2 and ember-cli-qunit version 4.0.1\r\n\r\nTo reiterate,  everything is working exactly as desired, it is only when running the tests in staging that I came across this problem. Is this a known issue or something someone else has been having? Or am I just looking in the wrong places for a solution?\r\n\r\nThanks for your time! Apologies if the explanation is a bit garbled, let me know if anything is unclear.\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16323","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16323/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16323/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16323/events","html_url":"https://github.com/emberjs/ember.js/issues/16323","id":302296359,"number":16323,"title":"Ember 3.0 - Failed assertion when JSON.stringify -ing an Ember Object ","user":{"login":"mani-mishra","id":3315240,"avatar_url":"https://avatars1.githubusercontent.com/u/3315240?v=4","gravatar_id":"","url":"https://api.github.com/users/mani-mishra","html_url":"https://github.com/mani-mishra","followers_url":"https://api.github.com/users/mani-mishra/followers","following_url":"https://api.github.com/users/mani-mishra/following{/other_user}","gists_url":"https://api.github.com/users/mani-mishra/gists{/gist_id}","starred_url":"https://api.github.com/users/mani-mishra/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/mani-mishra/subscriptions","organizations_url":"https://api.github.com/users/mani-mishra/orgs","repos_url":"https://api.github.com/users/mani-mishra/repos","events_url":"https://api.github.com/users/mani-mishra/events{/privacy}","received_events_url":"https://api.github.com/users/mani-mishra/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-03-05T13:18:09Z","updated_at":"2018-03-09T17:09:47Z","closed_at":"2018-03-09T17:09:47Z","author_association":"CONTRIBUTOR","body":"Upgrading to Ember 3.0 and using ```JSON.stringify```, results in following failed assertion\r\n\r\n```Assertion Failed: You attempted to access `someValue.toJSON` (on `<(unknown mixin):ember2486>`), but `someValue` is a computed property.↵↵Due to certain internal implementation details of Ember, ```\r\n\r\nI read the linked [RFCs and discussion](https://github.com/emberjs/rfcs/pull/281) and I understand this is just a precautionary 
8000
assertion, but this breaks ```JSON.stringify``` usage. \r\n\r\nhttps://github.com/douglascrockford/JSON-js/blob/master/json2.js#L237-L238\r\n\r\n\r\n<img width=\"1032\" alt=\"screen shot 2018-03-05 at 6 48 09 pm\" src=\"https://user-images.githubusercontent.com/3315240/36976993-939f615c-20a5-11e8-925d-ee1086763043.png\">\r\n\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16322","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16322/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16322/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16322/events","html_url":"https://github.com/emberjs/ember.js/issues/16322","id":302282402,"number":16322,"title":"Unhelpful syntax error message","user":{"login":"GuillaumeGomez","id":3050060,"avatar_url":"https://avatars0.githubusercontent.com/u/3050060?v=4","gravatar_id":"","url":"https://api.github.com/users/GuillaumeGomez","html_url":"https://github.com/GuillaumeGomez","followers_url":"https://api.github.com/users/GuillaumeGomez/followers","following_url":"https://api.github.com/users/GuillaumeGomez/following{/other_user}","gists_url":"https://api.github.com/users/GuillaumeGomez/gists{/gist_id}","starred_url":"https://api.github.com/users/GuillaumeGomez/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/GuillaumeGomez/subscriptions","organizations_url":"https://api.github.com/users/GuillaumeGomez/orgs","repos_url":"https://api.github.com/users/GuillaumeGomez/repos","events_url":"https://api.github.com/users/GuillaumeGomez/events{/privacy}","received_events_url":"https://api.github.com/users/GuillaumeGomez/received_events","type":"User","site_admin":false},"labels":[{"id":187302,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Bug","name":"Bug","color":"e10c02","default":false},{"id":259289356,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Help%20Wanted","name":"Help Wanted","color":"009800","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-03-05T12:32:20Z","updated_at":"2018-03-08T14:40:00Z","closed_at":"2018-03-08T14:40:00Z","author_association":"NONE","body":"I just updated to ember 3 and got the following error (everything else is working fine):\r\n\r\n```\r\nUncaught Error: SYNTAX ERROR: #if requires a single argument\r\n```\r\n\r\nConsidering the size of the codebase, I can't figure out where this error comes from (already looked at all `#if` so I suppose I missed one...) and didn't find out how or where it was generated in this repository so I can't send a PR either.\r\n\r\nCould you add the full `#if` syntax please? It'd be really helpful...","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16318","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16318/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16318/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16318/events","html_url":"https://github.com/emberjs/ember.js/issues/16318","id":302106645,"number":16318,"title":"Question: Router passing wrong model to setupController","user":{"login":"zion03","id":1479402,"avatar_url":"https://avatars0.githubusercontent.com/u/1479402?v=4","gravatar_id":"","url":"https://api.github.com/users/zion03","html_url":"https://github.com/zion03","followers_url":"https://api.github.com/users/zion03/followers","following_url":"https://api.github.com/users/zion03/following{/other_user}","gists_url":"https://api.github.com/users/zion03/gists{/gist_id}","starred_url":"https://api.github.com/users/zion03/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/zion03/subscriptions","organizations_url":"https://api.github.com/users/zion03/orgs","repos_url":"https://api.github.com/users/zion03/repos","events_url":"https://api.github.com/users/zion03/events{/privacy}","received_events_url":"https://api.github.com/users/zion03/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-03-04T18:02:45Z","updated_at":"2018-03-04T19:39:40Z","closed_at":"2018-03-04T19:39:40Z","author_association":"NONE","body":"When pass `model` to `link-to` it's goes directly to `setupController` and sometimes confused.\r\nIn my case:\r\n\r\n\r\n```\r\n  model: function (params) {\r\n    return RSVP.hash({\r\n      'category': this.store.peekRecord('category', params.id)\r\n    });\r\n  }\r\n```\r\n\r\n`{{#link-to 'category'  1}}Title{{/link-to}}` --> `model.category`\r\n`{{#link-to 'category'  model.id}}Title{{/link-to}}` --> `model.category`\r\n`{{#link-to 'category'  model}}Title{{/link-to}}`  --> `model` (only single model)\r\n\r\nSo it's break my templates where I use `model.category`\r\n\r\nIs it a normal behavior?","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16304","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16304/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16304/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16304/events","html_url":"https://github.com/emberjs/ember.js/issues/16304","id":301093590,"number":16304,"title":"2.18.2 missing from changelog","user":{"login":"jaswilli","id":214142,"avatar_url":"https://avatars1.githubusercontent.com/u/214142?v=4","gravatar_id":"","url":"https://api.github.com/users/jaswilli","html_url":"https://github.com/jaswilli","followers_url":"https://api.github.com/users/jaswilli/followers","following_url":"https://api.github.com/users/jaswilli/following{/other_user}","gists_url":"https://api.github.com/users/jaswilli/gists{/gist_id}","starred_url":"https://api.github.com/users/jaswilli/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/jaswilli/subscriptions","organizations_url":"https://api.github.com/users/jaswilli/orgs","repos_url":"https://api.github.com/users/jaswilli/repos","events_url":"https://api.github.com/users/jaswilli/events{/privacy}","received_events_url":"https://api.github.com/users/jaswilli/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-02-28T16:33:45Z","updated_at":"2018-03-05T20:01:34Z","closed_at":"2018-03-05T20:01:34Z","author_association":"CONTRIBUTOR","body":"Title says it all. 😄 ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16296","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16296/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16296/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16296/events","html_url":"https://github.com/emberjs/ember.js/issues/16296","id":300534128,"number":16296,"title":"Runloop can drop events","user":{"login":"ef4","id":319282,"avatar_url":"https://avatars0.githubusercontent.com/u/319282?v=4","gravatar_id":"","url":"https://api.github.com/users/ef4","html_url":"https://github.com/ef4","followers_url":"https://api.github.com/users/ef4/followers","following_url":"https://api.github.com/users/ef4/following{/other_user}","gists_url":"https://api.github.com/users/ef4/gists{/gist_id}","starred_url":"https://api.github.com/users/ef4/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ef4/subscriptions","organizations_url":"https://api.github.com/users/ef4/orgs","repos_url":"https://api.github.com/users/ef4/repos","events_url":"https://api.github.com/users/ef4/events{/privacy}","received_events_url":"https://api.github.com/users/ef4/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-02-27T08:29:36Z","updated_at":"2018-02-27T14:57:27Z","closed_at":"2018-02-27T14:54:14Z","author_association":"CONTRIBUTOR","body":"Hello fans of timing bugs, today I have a fun one for you.\r\n\r\nThe new microtask-based autorun architecture can drop scheduled events.  To see it in action, you can run the following against 3.1 beta:\r\n\r\n```js\r\nimport Controller from '@ember/controller';\r\nimport { schedule } from '@ember/runloop';\r\n\r\nexport default Controller.extend({\r\n  actions: {\r\n    testIt() {\r\n      // Here I am deliberately escaping the run loop to use autoruns instead.\r\n      Promise.resolve().then(() => {\r\n        // This schedules an autorun\r\n        schedule('render', () => {\r\n          console.log(1);\r\n          // Here we schedule a microtask. It will be ahead of the runloop's own next flush.\r\n          Promise.resolve().then(() => {\r\n            console.log(2);\r\n            return Promise.resolve().then(() => {\r\n              // This runs fine\r\n              console.log(3);\r\n              // But this scheduled event never fires.\r\n              schedule('actions', () => console.log('THIS NEVER RUNS!!!!') )\r\n            });\r\n          })\r\n          // when we return out of here, the runloop goes back to the \r\n          // first queue to start flushing again. But our promise stack is\r\n          // interleaving with the runloop's one-microtask-per-queue.\r\n        });\r\n      });\r\n    }\r\n  }\r\n});\r\n\r\n```\r\n\r\nIf you fire the `testIt` event, the console logs\r\n\r\n```\r\n1\r\n2\r\n3\r\n```\r\n\r\nThe number of layers of promises required to experience the bug depends on which queue you're trying to schedule. The runloop spends one microtask per queue, so our stack of promise resolutions is interleaved with each queue's flush. In the above example, we schedule on the `actions` queue immediately after that queue has just flushed, and the runloop never notices.\r\n\r\nYou can make it drop events on any other queue just by varying the number of promises you resolve in between the time you first trigger an autorun and the time you schedule your doomed event.\r\n\r\nI published a working example of the bug here: https://github.com/ef4/bug-repro/tree/bad-timing\r\n\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16292","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16292/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16292/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16292/events","html_url":"https://github.com/emberjs/ember.js/issues/16292","id":300192330,"number":16292,"title":"observers don't trigger in complex app","user":{"login":"BryanCrotaz","id":2016523,"avatar_url":"https://avatars2.githubusercontent.com/u/2016523?v=4","gravatar_id":"","url":"https://api.github.com/users/BryanCrotaz","html_url":"https://github.com/BryanCrotaz","followers_url":"https://api.github.com/users/BryanCrotaz/followers","following_url":"https://api.github.com/users/BryanCrotaz/following{/other_user}","gists_url":"https://api.github.com/users/BryanCrotaz/gists{/gist_id}","starred_url":"https://api.github.com/users/BryanCrotaz/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/BryanCrotaz/subscriptions","organizations_url":"https://api.github.com/users/BryanCrotaz/orgs","repos_url":"https://api.github.com/users/BryanCrotaz/repos","events_url":"https://api.github.com/users/BryanCrotaz/events{/privacy}","received_events_url":"https://api.github.com/users/BryanCrotaz/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":37,"created_at":"2018-02-26T10:55:28Z","updated_at":"2018-02-28T05:48:23Z","closed_at":"2018-02-27T11:49:20Z","author_association":"NONE","body":"OK, this is really weird. I have a complex app with lots of addons involved where no observers are triggering. If I create a simple test route with a simple observer watching a plain string property it doesn't fire.  If I put that same code into a new application it works every time.\r\n\r\nUsing `ember-source 2.16.2, 2.16.4, 2.18.2`\r\n\r\nThis change (commenting the else) to `ember-metal/lib/property_events` fixes the symptoms.\r\n```\r\nfunction notifyObservers(obj, keyName, meta$$1) {\r\n    if (meta$$1.isSourceDestroying()) {\r\n      return;\r\n    }\r\n\r\n    var eventName = keyName + ':change';\r\n    var listeners = void 0;\r\n    if (deferred > 0) {\r\n      listeners = observerSet.add(obj, keyName, eventName);\r\n      accumulateListeners(obj, eventName, listeners, meta$$1);\r\n    }// else {\r\n      sendEvent(obj, eventName, [obj, keyName]);\r\n    //}\r\n  }\r\n```\r\n\r\nWhat could I have broken in my code that would cause this fix to work?\r\n\r\nOr have I uncovered an edge case around Ember observers?","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16275","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16275/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16275/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16275/events","html_url":"https://github.com/emberjs/ember.js/issues/16275","id":299805772,"number":16275,"title":"META ISSUE: Routing fails to identify willBeActive for param-only changes.","user":{"login":"nathanhammond","id":20542,"avatar_url":"https://avatars1.githubusercontent.com/u/20542?v=4","gravatar_id":"","url":"https://api.github.com/users/nathanhammond","html_url":"https://github.com/nathanhammond","followers_url":"https://api.github.com/users/nathanhammond/followers","following_url":"https://api.github.com/users/nathanhammond/following{/other_user}","gists_url":"https://api.github.com/users/nathanhammond/gists{/gist_id}","starred_url":"https://api.github.com/users/nathanhammond/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/nathanhammond/subscriptions","organizations_url":"https://api.github.com/users/nathanhammond/orgs","repos_url":"https://api.github.com/users/nathanhammond/repos","events_url":"https://api.github.com/users/nathanhammond/events{/privacy}","received_events_url":"https://api.github.com/users/nathanhammond/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":0,"created_at":"2018-02-23T18:25:08Z","updated_at":"2018-03-07T23:00:03Z","closed_at":"2018-03-07T23:00:03Z","author_association":"MEMBER","body":"There are numerous issues reported on this topic. I'm going to fix them because I need it to work correctly.\r\n\r\nSimplified reproduction as a Twiddle with instructions: https://ember-twiddle.com/52669dbb90a5f099cd6e78957e9fd892?openFiles=routes.products.details.js%2C&route=%2Fproducts%2F4\r\n\r\nOne piece of new information: it only fails for async transitions.\r\n\r\nThis is the line that triggers the bug:\r\nhttps://github.com/tildeio/router.js/blame/17a4cd3b6ac8053f3eb20d4358e2bc37653774b6/lib/router/transition-intent/named-transition-intent.js#L121\r\n\r\nThis has been existing since *at least* 2.0.\r\n\r\nRelated:\r\n- [ ] https://github.com/emberjs/ember.js/issues/12419\r\n- [ ] https://github.com/emberjs/ember.js/issues/10989\r\n- [ ] https://github.com/emberjs/ember.js/issues/10195\r\n- [ ] https://github.com/tildeio/router.js/issues/236\r\n\r\n/cc @topaxi @bgentry ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16269","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16269/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16269/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16269/events","html_url":"https://github.com/emberjs/ember.js/issues/16269","id":299465313,"number":16269,"title":"Operand over 16-bits. Got 72000.","user":{"login":"Turbo87","id":141300,"avatar_url":"https://avatars2.githubusercontent.com/u/141300?v=4","gravatar_id":"","url":"https://api.github.com/users/Turbo87","html_url":"https://github.com/Turbo87","followers_url":"https://api.github.com/users/Turbo87/followers","following_url":"https://api.github.com/users/Turbo87/following{/other_user}","gists_url":"https://api.github.com/users/Turbo87/gists{/gist_id}","starred_url":"https://api.github.com/users/Turbo87/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Turbo87/subscriptions","organizations_url":"https://api.github.com/users/Turbo87/orgs","repos_url":"https://api.github.com/users/Turbo87/repos","events_url":"https://api.github.com/users/Turbo87/events{/privacy}","received_events_url":"https://api.github.com/users/Turbo87/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":4,"created_at":"2018-02-22T18:47:32Z","updated_at":"2018-02-25T15:51:03Z","closed_at":"2018-02-25T15:51:03Z","author_association":"MEMBER","body":"Using the latest 3.1 beta version I'm seeing an error \"Operand over 16-bits. Got 72000.\" in one of our tests which does this:\r\n\r\n```\r\nrender(hbs`{{some-component foo=9000}}`);\r\n```\r\n\r\nIt seems that the maximum value for literal numbers has decreased significantly in the 3.1-beta 🤔 \r\n\r\nhttps://github.com/emberjs/ember.js/issues/15635 might also be related to this.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16267","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16267/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16267/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16267/events","html_url":"https://github.com/emberjs/ember.js/issues/16267","id":298972923,"number":16267,"title":"Usage of testem.js and ember.js with two different environments (ci, dev)","user":{"login":"brunoocasali","id":4116980,"avatar_url":"https://avatars1.githubusercontent.com/u/4116980?v=4","gravatar_id":"","url":"https://api.github.com/users/brunoocasali","html_url":"https://github.com/brunoocasali","followers_url":"https://api.github.com/users/brunoocasali/followers","following_url":"https://api.github.com/users/brunoocasali/following{/other_user}","gists_url":"https://api.github.com/users/brunoocasali/gists{/gist_id}","starred_url":"https://api.github.com/users/brunoocasali/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/brunoocasali/subscriptions","organizations_url":"https://api.github.com/users/brunoocasali/orgs","repos_url":"https://api.github.com/users/brunoocasali/repos","events_url":"https://api.github.com/users/brunoocasali/events{/privacy}","received_events_url":"https://api.github.com/users/brunoocasali/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-02-21T13:38:04Z","updated_at":"2018-02-24T03:59:02Z","closed_at":"2018-02-24T03:59:02Z","author_association":"NONE","body":"Hello and thank you for the awesome project\r\nPlease correct me if this is a testemjs issue or other...\r\n\r\nI'm using the ember-cli 2.18. And I'm on a macOS Sierra.\r\n\r\nMy testem.js is:\r\n```js\r\nmodule.exports = {\r\n  test_page: 'tests/index.html?hidepassed&nocontainer',\r\n  disable_watching: true,\r\n  parallel: 4,\r\n  launch_in_ci: [\r\n    'chromium'\r\n  ],\r\n  launch_in_dev: [\r\n    'Chrome'\r\n  ],\r\n  browser_args: {\r\n    Chrome: {\r\n      mode: 'dev',\r\n      args: [\r\n        '--headless',\r\n        '--disable-gpu',\r\n        '--no-sandbox',\r\n        '--remote-debugging-port=0',\r\n        '--window-size=1440,900',\r\n        '--no-default-browser-check',\r\n        '--no-first-run',\r\n        '--ignore-certificate-errors',\r\n        '--test-type',\r\n        '--disable-renderer-backgrounding',\r\n        '--disable-background-timer-throttling'\r\n      ]\r\n    },\r\n    chromium: {\r\n      mode: 'ci',\r\n      args: [\r\n        '--headless',\r\n        '--disable-gpu',\r\n        '--no-sandbox',\r\n        '--remote-debugging-port=0',\r\n        '--window-size=1440,900',\r\n        '--no-default-browser-check',\r\n        '--no-first-run',\r\n        '--ignore-certificate-errors',\r\n        '--test-type',\r\n        '--disable-renderer-backgrounding',\r\n        '--disable-background-timer-throttling',\r\n      ]\r\n    }\r\n  }\r\n};\r\n```\r\nHow to run the tests in the development mode? I have no chromium installed on macOS but yes in GitlabCI.\r\n\r\n```zsh\r\n➜  dash git:(enhancement/upgrade-ember-version) ✗ ember test\r\ncleaning up...\r\nBuilt project successfully. Stored in \"/Users/bruno/develop/alpha/dash/tmp/class-tests_dist-18xhIjJw.tmp\".\r\nnot ok 1 Error\r\n    ---\r\n        message: >\r\n            Launcher chromium not found. Not installed?\r\n    ...\r\n\r\n1..1\r\n# tests 1\r\n# pass  0\r\n# skip  0\r\n# fail  1\r\nLauncher chromium not found. Not installed?\r\n\r\n\r\nStack Trace and Error Report: /var/folders/z5/sql4q4qx12s9vl098z8gyffm0000gn/T/error.dump.3ab25e243b2f52d7c5419f71452d9a6a.log\r\n➜  dash git:(enhancement/upgrade-ember-version) ✗\r\n```\r\n\r\nToday I need to run tests passing some flags: `ember test --launch Chrome --server` not just the command `ember test`, and this kind of usage seems tricky to me... \r\n\r\nThis is correct or I'm missing something?\r\n\r\nPS: At GitLab CI, I use normally the `yarn test` wich resolves to `ember test` only.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16263","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16263/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16263/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16263/events","html_url":"https://github.com/emberjs/ember.js/issues/16263","id":298578888,"number":16263,"title":"You attempted to access the `__DESCRIPTOR__` property (of <frontend@service:prototypes::ember581>)","user":{"login":"sly7-7","id":1826661,"avatar_url":"https://avatars1.githubusercontent.com/u/1826661?v=4","gravatar_id":"","url":"https://api.github.com/users/sly7-7","html_url":"https://github.com/sly7-7","followers_url":"https://api.github.com/users/sly7-7/followers","following_url":"https://api.github.com/users/sly7-7/following{/other_user}","gists_url":"https://api.github.com/users/sly7-7/gists{/gist_id}","starred_url":"https://api.github.com/users/sly7-7/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/sly7-7/subscriptions","organizations_url":"https://api.github.com/users/sly7-7/orgs","repos_url":"https://api.github.com/users/sly7-7/repos","events_url":"https://api.github.com/users/sly7-7/events{/privacy}","received_events_url":"https://api.github.com/users/sly7-7/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":7,"created_at":"2018-02-20T12:38:07Z","updated_at":"2018-02-24T18:44:02Z","closed_at":"2018-02-23T17:55:37Z","author_association":"CONTRIBUTOR","body":"Related to #16148\r\n\r\nI have a service, injected in a route (extending the AuthenticatedRouteMixin). When I do `this.get('prototypes')`,so, in my use case I don't know if the error is coming from: \r\n * The route accessing a service\r\n * The route itself\r\n * The service itself\r\n\r\nLet me now if you need more information.\r\nFor now, I will try to revert 2.18 (ember 3.0.X seems to have problems with ie 11)\r\n\r\n```\r\nThere is more information at https://github.com/emberjs/ember.js/issues/16148, and\r\nyou can help us improve this error message by telling us more about what happened in\r\nthis situation.\r\n    at new EmberError (error.js:40)\r\n    at assert (index.js:139)\r\n    at Object.get (core_object.js:57)\r\n    at isDescriptorTrap (ember-metal.js:3258)\r\n    at get (ember-metal.js:3430)\r\n    at Class.get (observable.js:38)\r\n    at route.js:62\r\n```\r\ncc/ @wycats \r\nMaybe this is an issue in ember-simple-auth, so kindly cc/ to @marcoow ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16259","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16259/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16259/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16259/events","html_url":"https://github.com/emberjs/ember.js/issues/16259","id":298329553,"number":16259,"title":"Assertion Failed: You attempted to access the `size` property","user":{"login":"Turbo87","id":141300,"avatar_url":"https://avatars2.githubusercontent.com/u/141300?v=4","gravatar_id":"","url":"https://api.github.com/users/Turbo87","html_url":"https://github.com/Turbo87","followers_url":"https://api.github.com/users/Turbo87/followers","following_url":"https://api.github.com/users/Turbo87/following{/other_user}","gists_url":"https://api.github.com/users/Turbo87/gists{/gist_id}","starred_url":"https://api.github.com/users/Turbo87/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Turbo87/subscriptions","organizations_url":"https://api.github.com/users/Turbo87/orgs","repos_url":"https://api.github.com/users/Turbo87/repos","events_url":"https://api.github.com/users/Turbo87/events{/privacy}","received_events_url":"https://api.github.com/users/Turbo87/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-02-19T16:10:43Z","updated_at":"2018-02-23T08:01:07Z","closed_at":"2018-02-23T08:01:07Z","author_association":"MEMBER","body":"\r\non Ember 3.1.0-beta.1:\r\n\r\n> Assertion Failed: You attempted to access the `size` property (of <DS.Errors:ember1800>).\r\nSince Ember 3.1, this is usually fine as you no longer need to use `.get()`\r\nto access computed properties. However, in this case, the object in question\r\nis a special kind of Ember object (a proxy). Therefore, it is still necessary\r\nto use `.get('size')` in this case.\r\n>\r\n> If you encountered this error because of third-party code that you don't control,\r\nthere is more information at https://github.com/emberjs/ember.js/issues/16148, and\r\nyou can help us improve this error message by telling us more about what happened in\r\nthis situation.\r\n\r\nThis seems to be caused by Ember Data using `isEmpty()` on a `DS.Errors` object. Since `isEmpty()` checks `obj.size` unconditionally it is causing the error above.\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16257","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16257/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16257/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16257/events","html_url":"https://github.com/emberjs/ember.js/issues/16257","id":298183303,"number":16257,"title":"Uncaught TypeError: RSVP.cast is not a function","user":{"login":"Turbo87","id":141300,"avatar_url":"https://avatars2.githubusercontent.com/u/141300?v=4","gravatar_id":"","url":"https://api.github.com/users/Turbo87","html_url":"https://github.com/Turbo87","followers_url":"https://api.github.com/users/Turbo87/followers","following_url":"https://api.github.com/users/Turbo87/following{/other_user}","gists_url":"https://api.github.com/users/Turbo87/gists{/gist_id}","starred_url":"https://api.github.com/users/Turbo87/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Turbo87/subscriptions","organizations_url":"https://api.github.com/users/Turbo87/orgs","repos_url":"https://api.github.com/users/Turbo87/repos","events_url":"https://api.github.com/users/Turbo87/events{/privacy}","received_events_url":"https://api.github.com/users/Turbo87/received_events","type":"User","site_admin":false},"labels":[{"id":394458286,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Has%20PR","name":"Has PR","color":"c5def5","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-02-19T07:35:07Z","updated_at":"2018-03-09T20:22:06Z","closed_at":"2018-03-09T20:22:06Z","author_association":"MEMBER","body":"It seems that Ember v3.1.0-beta.1 silently updated RSVP to a version that no longer includes `RSVP.cast()`, but unfortunately there was no deprecation warning prior to this. `ember-intl` seems to use that function internally though which broke our tests :-/\r\n\r\n/cc @jasonmit ","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16256","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16256/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16256/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16256/events","html_url":"https://github.com/emberjs/ember.js/issues/16256","id":298035423,"number":16256,"title":"Input helper passed type does not work in Ember 3.1 beta","user":{"login":"rwwagner90","id":2640861,"avatar_url":"https://avatars3.githubusercontent.com/u/2640861?v=4","gravatar_id":"","url":"https://api.github.com/users/rwwagner90","html_url":"https://github.com/rwwagner90","followers_url":"https://api.github.com/users/rwwagner90/followers","following_url":"https://api.github.com/users/rwwagner90/following{/other_user}","gists_url":"https://api.github.com/users/rwwagner90/gists{/gist_id}","starred_url":"https://api.github.com/users/rwwagner90/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rwwagner90/subscriptions","organizations_url":"https://api.github.com/users/rwwagner90/orgs","repos_url":"https://api.github.com/users/rwwagner90/repos","events_url":"https://api.github.com/users/rwwagner90/events{/privacy}","received_events_url":"https://api.github.com/users/rwwagner90/received_events","type":"User","site_admin":false},"labels":[{"id":394458286,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Has%20PR","name":"Has PR","color":"c5def5","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":5,"created_at":"2018-02-17T20:52:57Z","updated_at":"2018-02-27T14:53:38Z","closed_at":"2018-02-27T13:49:28Z","author_association":"MEMBER","body":"I was working on updating my apps and addons to 3.1 beta and noticed some strange behavior. I have components that wrap the `{{input}}` helper and only the first one was rendering an input. I created a reproduction here https://github.com/rwwagner90/inputs and I played around with it some. It appears the issue is passing `type` down as an argument. It chokes on:\r\n\r\n```hbs\r\n{{input type=type}}\r\n```\r\n\r\nI also tested this against https://s3.amazonaws.com/builds.emberjs.com/canary/shas/9300d1b7d3aa70987986b1997ecf02a542ea49be.tgz and there did not seem to be any difference. \r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16247","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16247/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16247/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16247/events","html_url":"https://github.com/emberjs/ember.js/issues/16247","id":297482130,"number":16247,"title":"transitionToRoute is gone from the API documentation","user":{"login":"cbou","id":611389,"avatar_url":"https://avatars0.githubusercontent.com/u/611389?v=4","gravatar_id":"","url":"https://api.github.com/users/cbou","html_url":"https://github.com/cbou","followers_url":"https://api.github.com/users
50ac
/cbou/followers","following_url":"https://api.github.com/users/cbou/following{/other_user}","gists_url":"https://api.github.com/users/cbou/gists{/gist_id}","starred_url":"https://api.github.com/users/cbou/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/cbou/subscriptions","organizations_url":"https://api.github.com/users/cbou/orgs","repos_url":"https://api.github.com/users/cbou/repos","events_url":"https://api.github.com/users/cbou/events{/privacy}","received_events_url":"https://api.github.com/users/cbou/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2018-02-15T15:15:49Z","updated_at":"2018-03-05T12:10:42Z","closed_at":"2018-03-05T09:46:32Z","author_association":"CONTRIBUTOR","body":"What happens to `transitionToRoute ` in the API Documentation?\r\n\r\nIt was in the 2.18.0 but not in anymore in the 3.0.0.\r\n\r\nhttps://emberjs.com/api/ember/2.18/classes/Controller\r\nhttps://emberjs.com/api/ember/release/classes/Controller","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16236","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16236/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16236/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16236/events","html_url":"https://github.com/emberjs/ember.js/issues/16236","id":296193560,"number":16236,"title":"one-way binding can't be re-rendered when coexisting with two-way binding","user":{"login":"nightire","id":1104018,"avatar_url":"https://avatars3.githubusercontent.com/u/1104018?v=4","gravatar_id":"","url":"https://api.github.com/users/nightire","html_url":"https://github.com/nightire","followers_url":"https://api.github.com/users/nightire/followers","following_url":"https://api.github.com/users/nightire/following{/other_user}","gists_url":"https://api.github.com/users/nightire/gists{/gist_id}","starred_url":"https://api.github.com/users/nightire/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/nightire/subscriptions","organizations_url":"https://api.github.com/users/nightire/orgs","repos_url":"https://api.github.com/users/nightire/repos","events_url":"https://api.github.com/users/nightire/events{/privacy}","received_events_url":"https://api.github.com/users/nightire/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":5,"created_at":"2018-02-11T14:21:18Z","updated_at":"2018-02-25T18:14:21Z","closed_at":"2018-02-23T14:40:48Z","author_association":"CONTRIBUTOR","body":"Reproduction twiddle:\r\n\r\nhttps://ember-twiddle.com/8ea355e29321eba666ec70ea821dd9ac\r\n\r\nThis demo shows a bug which\r\n\r\na component takes an one-way binding value and a two-way binding value at the same time, when trying to mutate both of them: \r\n\r\n1. two-way binding value can be updated and re-rendered\r\n2. one-way binding value can be updated (logged in console), but cannot be re-rendered properly\r\n\r\nHowever, trying to mutate one-way binding value only doesn't have any problem.\r\n\r\n---\r\n\r\nP.S. I'd like to send a PR to provide a test case, but I'm not sure what kind of tests should be write. It would be helpful if someone can show me an example.","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16219","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16219/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16219/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16219/events","html_url":"https://github.com/emberjs/ember.js/issues/16219","id":294899421,"number":16219,"title":"ember-simple-auth demo not working for me","user":{"login":"MarshallSchaffer","id":32654358,"avatar_url":"https://avatars0.githubusercontent.com/u/32654358?v=4","gravatar_id":"","url":"https://api.github.com/users/MarshallSchaffer","html_url":"https://github.com/MarshallSchaffer","followers_url":"https://api.github.com/users/MarshallSchaffer/followers","following_url":"https://api.github.com/users/MarshallSchaffer/following{/other_user}","gists_url":"https://api.github.com/users/MarshallSchaffer/gists{/gist_id}","starred_url":"https://api.github.com/users/MarshallSchaffer/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/MarshallSchaffer/subscriptions","organizations_url":"https://api.github.com/users/MarshallSchaffer/orgs","repos_url":"https://api.github.com/users/MarshallSchaffer/repos","events_url":"https://api.github.com/users/MarshallSchaffer/events{/privacy}","received_events_url":"https://api.github.com/users/MarshallSchaffer/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":2,"created_at":"2018-02-06T20:18:39Z","updated_at":"2018-02-07T02:15:22Z","closed_at":"2018-02-07T02:09:53Z","author_association":"NONE","body":"Hi.  Yesterday, I spent hours trying to install ember-simple-auth.  I want to see how the demo works.\r\n\r\nI am using the following versions\r\n\r\nEmber 2.18, node v8.9.4, npm 5.6.0, git version 2.16.1.windows.2, eslint v4.17.0, the current versions of qunit and mocha, yarn, etc. etc.\r\n\r\nI followed the instruction on https://github.com/simplabs/ember-simple-auth#readme   , beginning with \r\n`git clone https://github.com/simplabs/ember-simple-auth.git`\r\n\r\nThe output was\r\n\r\nC:\\Javascript\\Ember7\\ember-simple-auth>yarn install\r\nyarn install v1.3.2\r\n[1/5] Validating package.json...\r\n[2/5] Resolving packages...\r\n[3/5] Fetching packages...\r\ninfo fsevents@1.1.3: The platform \"win32\" is incompatible with this module.\r\ninfo \"fsevents@1.1.3\" is an optional dependency and failed compatibility check. Excluding it from installation.\r\n[4/5] Linking dependencies...\r\nwarning \" > eslint-config-simplabs@0.4.0\" has unmet peer dependency \"eslint@>= 3\".\r\nwarning \" > eslint-plugin-mocha@4.11.0\" has unmet peer dependency \"eslint@^2.0.0 || ^3.0.0 || ^4.0.0\".\r\nwarning \" > sinon-chai@2.14.0\" has unmet peer dependency \"sinon@^1.4.0 || ^2.1.0 || ^3.0.0 || ^4.0.0\".\r\n[5/5] Building fresh packages...\r\nsuccess Saved lockfile.\r\n\r\nBut when I tried to do \"ember serve\", I got\r\n\r\nC:\\Javascript\\Ember7\\ember-simple-auth>ember s\r\nDEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require(\"resolve\").sync(something, { basedir: project.root }). Accessed from:   new NPMDependencyVersionChecker (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli-version-checker\\src\\npm-dependency-version-checker.js:11:33)\r\nWARNING: ember-decorators (used in ember-popper): You have not installed @ember-decorators/babel-transforms. It has been extracted to a separate addon. See instructions for installation: https://github.com/ember-decorators/babel-transforms#readme\r\nember-getowner-polyfill is not required for Ember 2.3.0 and later, please remove from your `package.json`.\r\nThe \"group\" options can only be used with a \"testGenerator\" option of: qunit,mocha\r\n**Error:** The \"group\" options can only be used with a \"testGenerator\" option of: qunit,mocha\r\n    at Function.EslintValidationFilter.create (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\broccoli-lint-eslint\\lib\\index.js:205:11)\r\n    at Class.lintTree (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli-eslint\\index.js:44:19)\r\n    at project.addons.reduce (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:428:26)\r\n    at Array.reduce (<anonymous>)\r\n    at Class._eachProjectAddonInvoke (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:425:32)\r\n    at Class.jshintAddonTree (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:1136:35)\r\n    at Class.treeFor (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:545:23)\r\n    at project.addons.reduce (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\broccoli\\ember-app.js:630:25)\r\n    at Array.reduce (<anonymous>)\r\n    at EmberAddon.addonTreesFor (C:\\Javascript\\Ember7\\ember-simple-auth\\node_modules\\ember-cli\\lib\\broccoli\\ember-app.js:628:32)\r\n\r\nSince then, I've tried installing @ember-decorators/babel-transforms, ESlint, qunit, mocha, broccoli.  \r\n\"Also,\" my attempts to install ember-decorators/babel-transforms fail, regard of whether or not I use the \"-S\" option.  I have tried this several times:\r\n\r\nWARNING: ember-decorators (used in ember-popper): You have not installed @ember-decorators/babel-transforms. It has been extracted to a separate addon. See instructions for installation: https://github.com/ember-decorators/babel-transforms#readme\r\n\r\nAfter updating \"everything,\" here is the LATEST OUTPUT:\r\n\r\nC:\\Javascript\\Ember8\\ember-simple-auth>ember s\r\nDEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require(\"resolve\").sync(something, { basedir: project.root }). Accessed from:   new NPMDependencyVersionChecker (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli-version-checker\\src\\npm-dependency-version-checker.js:11:33)\r\nWARNING: ember-decorators (used in ember-popper): You have not installed @ember-decorators/babel-transforms. It has been extracted to a separate addon. See instructions for installation: https://github.com/ember-decorators/babel-transforms#readme\r\nThe \"group\" options can only be used with a \"testGenerator\" option of: qunit,mocha\r\nError: The \"group\" options can only be used with a \"testGenerator\" option of: qunit,mocha\r\n    at Function.EslintValidationFilter.create (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\broccoli-lint-eslint\\lib\\index.js:205:11)\r\n    at Class.lintTree (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli-eslint\\index.js:44:19)\r\n    at project.addons.reduce (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:428:26)\r\n    at Array.reduce (<anonymous>)\r\n    at Class._eachProjectAddonInvoke (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:425:32)\r\n    at Class.jshintAddonTree (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:1136:35)\r\n    at Class.treeFor (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\models\\addon.js:545:23)\r\n    at project.addons.reduce (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\broccoli\\ember-app.js:630:25)\r\n    at Array.reduce (<anonymous>)\r\n    at EmberAddon.addonTreesFor (C:\\Javascript\\Ember8\\ember-simple-auth\\node_modules\\ember-cli\\lib\\broccoli\\ember-app.js:628:32)\r\n\r\nI don't feel like I am getting anywhere.  I'd appreciate some advice.  Thank you.\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16201","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16201/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16201/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16201/events","html_url":"https://github.com/emberjs/ember.js/issues/16201","id":293161275,"number":16201,"title":"Don't set meta data on frozen objects","user":{"login":"broerse","id":1432898,"avatar_url":"https://avatars3.githubusercontent.com/u/1432898?v=4","gravatar_id":"","url":"https://api.github.com/users/broerse","html_url":"https://github.com/broerse","followers_url":"https://api.github.com/users/broerse/followers","following_url":"https://api.github.com/users/broerse/following{/other_user}","gists_url":"https://api.github.com/users/broerse/gists{/gist_id}","starred_url":"https://api.github.com/users/broerse/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/broerse/subscriptions","organizations_url":"https://api.github.com/users/broerse/orgs","repos_url":"https://api.github.com/users/broerse/repos","events_url":"https://api.github.com/users/broerse/events{/privacy}","received_events_url":"https://api.github.com/users/broerse/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":5,"created_at":"2018-01-31T13:40:05Z","updated_at":"2018-02-14T11:58:57Z","closed_at":"2018-02-14T07:00:52Z","author_association":"NONE","body":"To fix `Only string, number, symbol, boolean, null, undefined, and function are allowed as default properties (ember/avoid-leaking-state-in-ember-objects)` for ember 2.18 I changed my code to: \r\n\r\n```javascript\r\n  postsSorting: Object.freeze(['date:desc']),\r\n  arrangedContent: sort('posts', 'postsSorting'),\r\n```\r\n\r\nBut then Ember stops working in IE 11. Perhaps Ember should not set meta data on frozen objects. The dutch warning I get `Kan de eigenschap __ember_meta__ niet definiëren: object is niet uitbreidbaar` that points to  `<-------------------------` \r\n\r\nBut can't find it in https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/meta.js\r\n\r\nIs it already fixed in Ember > 2.18 ?\r\n\r\n```javascript\r\n  var setMeta = void 0;\r\n  exports.peekMeta = void 0;\r\n\r\n  // choose the one appropriate for given platform\r\n  if (emberUtils.HAS_NATIVE_WEAKMAP) {\r\n    var getPrototypeOf = Object.getPrototypeOf;\r\n    var metaStore = new WeakMap();\r\n\r\n    setMeta = function WeakMap_setMeta(obj, meta) {\r\n      {\r\n        counters.setCalls++;\r\n      }\r\n      metaStore.set(obj, meta);\r\n    };\r\n\r\n    exports.peekMeta = function WeakMap_peekParentMeta(obj) {\r\n      var pointer = obj;\r\n      var meta = void 0;\r\n      while (pointer !== undefined && pointer !== null) {\r\n        meta = metaStore.get(pointer);\r\n        // jshint loopfunc:true\r\n        {\r\n          counters.peekCalls++;\r\n        }\r\n        if (meta !== undefined) {\r\n          return meta;\r\n        }\r\n\r\n        pointer = getPrototypeOf(pointer);\r\n        {\r\n          counters.peekPrototypeWalks++;\r\n        }\r\n      }\r\n    };\r\n  } else {\r\n    setMeta = function Fallback_setMeta(obj, meta) {\r\n      if (obj.__defineNonEnumerable) {\r\n        obj.__defineNonEnumerable(EMBER_META_PROPERTY);\r\n      } else {\r\n        Object.defineProperty(obj, META_FIELD, META_DESC);   <-------------------------\r\n      }\r\n\r\n      obj[META_FIELD] = meta;\r\n    };\r\n\r\n    exports.peekMeta = function Fallback_peekMeta(obj) {\r\n      return obj[META_FIELD];\r\n    };\r\n  }\r\n```","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16198","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16198/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16198/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16198/events","html_url":"https://github.com/emberjs/ember.js/issues/16198","id":292725972,"number":16198,"title":"yarn test fails with '/' in branch name.","user":{"login":"Parrryy","id":7785538,"avatar_url":"https://avatars1.githubusercontent.com/u/7785538?v=4","gravatar_id":"","url":"https://api.github.com/users/Parrryy","html_url":"https://github.com/Parrryy","followers_url":"https://api.github.com/users/Parrryy/followers","following_url":"https://api.github.com/users/Parrryy/following{/other_user}","gists_url":"https://api.github.com/users/Parrryy/gists{/gist_id}","starred_url":"https://api.github.com/users/Parrryy/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/Parrryy/subscriptions","organizations_url":"https://api.github.com/users/Parrryy/orgs","repos_url":"https://api.github.com/users/Parrryy/repos","events_url":"https://api.github.com/users/Parrryy/events{/privacy}","received_events_url":"https://api.github.com/users/Parrryy/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-01-30T10:17:03Z","updated_at":"2018-02-05T00:31:40Z","closed_at":"2018-02-05T00:31:40Z","author_association":"CONTRIBUTOR","body":"When running `yarn test` with a '/' in the branch name the test fails with the following message.\r\n\r\n```\r\nTest failed: ember-metal/core/main: Ember.VERSION is in alignment with SemVer v2.0.0\r\n    Failed assertion: Ember.VERSION (3.1.0-fix/ember-glimmer-views-doc-update+ca894ce2)is valid SemVer v2.0.0, expected: true, but was: false\r\n    at Object.<anonymous> (http://localhost:13141/ember-tests.js:42211:12)\r\n    at runTest (http://localhost:13141/qunit/qunit.js:1478:30)\r\n    at Test.run (http://localhost:13141/qunit/qunit.js:1464:6)\r\n    at http://localhost:13141/qunit/qunit.js:1670:12\r\n    at Object.advance (http://localhost:13141/qunit/qunit.js:1116:26)\r\n    at begin (http://localhost:13141/qunit/qunit.js:3059:20)\r\n```\r\n\r\nIt should convert illegal characters in the branch name to something legal.\r\n","score":1.0},{"url":"https://api.github.com/repos/emberjs/ember.js/issues/16190","repository_url":"https://api.github.com/repos/emberjs/ember.js","labels_url":"https://api.github.com/repos/emberjs/ember.js/issues/16190/labels{/name}","comments_url":"https://api.github.com/repos/emberjs/ember.js/issues/16190/comments","events_url":"https://api.github.com/repos/emberjs/ember.js/issues/16190/events","html_url":"https://github.com/emberjs/ember.js/issues/16190","id":292278888,"number":16190,"title":"Documentation for Ember.compare needs to be updated for rfc 176","user":{"login":"toddjordan","id":3609063,"avatar_url":"https://avatars3.githubusercontent.com/u/3609063?v=4","gravatar_id":"","url":"https://api.github.com/users/toddjordan","html_url":"https://github.com/toddjordan","followers_url":"https://api.github.com/users/toddjordan/followers","following_url":"https://api.github.com/users/toddjordan/following{/other_user}","gists_url":"https://api.github.com/users/toddjordan/gists{/gist_id}","starred_url":"https://api.github.com/users/toddjordan/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/toddjordan/subscriptions","organizations_url":"https://api.github.com/users/toddjordan/orgs","repos_url":"https://api.github.com/users/toddjordan/repos","events_url":"https://api.github.com/users/toddjordan/events{/privacy}","received_events_url":"https://api.github.com/users/toddjordan/received_events","type":"User","site_admin":false},"labels":[{"id":203287,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Documentation","name":"Documentation","color":"fbca04","default":false},{"id":73481071,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Good%20for%20New%20Contributors","name":"Good for New Contributors","color":"abcdef","default":false},{"id":259289356,"url":"https://api.github.com/repos/emberjs/ember.js/labels/Help%20Wanted","name":"Help Wanted","color":"009800","default":false}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-01-29T04:06:42Z","updated_at":"2018-01-29T04:10:24Z","closed_at":"2018-01-29T04:10:23Z","author_association":"CONTRIBUTOR","body":"In `ember-runtime/lib/compare.js`:\r\n```javascript\r\n/*\r\n...\r\n @method compare\r\n @for Ember\r\n @param {Object} v First value to compare\r\n @param {Object} w Second value to compare\r\n @return {Number} -1 if v < w, 0 if v = w and 1 if v > w.\r\n @public\r\n*/\r\n```\r\n\r\nNeeds to be updated to rfc 176\r\n```javascript\r\n/**\r\n@module @ember/utils\r\n*/\r\n/*\r\n...\r\n @method compare\r\n @for @ember/utils\r\n @static\r\n @param {Object} v First value to compare\r\n @param {Object} w Second value to compare\r\n @return {Number} -1 if v < w, 0 if v = w and 1 if v > w.\r\n @public\r\n*/\r\n```\r\n\r\nThe examples in the documentation need to be updated as well.","score":1.0}]}
0

